ladybug/datatype/base.py

```diff
@@ -44,25 +46,65 @@ class DataTypeBase(object):
         missing_epw: Missing value for the data type when it occurs in EPW files.
             (Default: None)
     """
+    _name = None
+    _units = [None]
+    _si_units = [None]
+    _ip_units = [None]
+    _min = float('-inf')
+    _max = float('+inf')
 
-    name = 'Data Type Base'
-    units = [None]
-    min = float('-inf')
-    max = float('+inf')
+    _abbreviation = ''
+    _unit_descr = ''
+    _point_in_time = True
+    _cumulative = False
 
-    abbreviation = ''
-    unit_descr = ''
-    point_in_time = True
-    cumulative = False
+    _min_epw = float('-inf')
+    _max_epw = float('+inf')
+    _missing_epw = None
 
-    min_epw = float('-inf')
-    max_epw = float('+inf')
-    missing_epw = None
+    def __init__(self, name=None):
+        """Initialize DataType.
 
-    def __init__(self):
-        """Init DataType."""
+        Args:
+            name: Optional name for the type. Default is derived from the class name.
+        """
+        self._name = name
+
+    @classmethod
+    def from_json(cls, data):
+        """Create a data type from a dictionary.
+
+        Args:
+            data: Data as a dictionary.
+                {
+                    "name": data type name of the data type as a string
+                    "class_name": the class name of the data type as a string
+                    "base_unit": the base unit of the data type
+                }
+        """
+        assert 'name' in data, 'Required keyword "name" is missing!'
+        assert 'class_name' in data, 'Required keyword "class_name" is missing!'
+
+        from ..datatype import _data_types
```

Comment:
I believe you have tested this but this looks like a circular import. datatype will import __init__ which will import _enumeration.py which will import base.py. Can we take the data that we need directly from base.py?

Reply:
I have now gotten rid of the circular import by putting the class that builds the enumeration in the same one as the DataTypeBase.